sdks/java/extensions/ordered/src/main/java/org/apache/beam/sdk/extensions/ordered/MutableState.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.ordered;
+
+import java.io.Serializable;
+
+/** Mutable state mutates when events apply to it. It is stored in a Beam state. */
+public interface MutableState<EventT, ResultT> extends Serializable {
+
+  /**
+   * The interface assumes that events will mutate the state without the possibility of throwing an
+   * error.
+   *
+   * <p>TODO: this might be too simplistic and a mechanism for failure of applying the event to a
+   * state would need to be created.
```

Comment:
My take: this is fine as is; in a batch pipeline, we'll fail the pipeline. In a streaming pipeline (more common use case), we'll keep retrying indefinitely and the user will need to take some action (and they can be alerted by monitoring OrderedProcessingStatus); that seems like the behavior we want here.

Users can also implement more complicated try/catch logic if they want.

I think this todo is fine to remove